cypress-io/cypress
# fix: restart Cypress server and browser on baseUrl change

- Closes #22056 

### User facing changelog


The `baseUrl` config option will now restart the Cypress server automatically when it is updated in a Cypress config file.

### Additional details


There are several things that use the baseUrl during the server's initialization like the [browser and xhr urls](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/config.ts#L532-L534) and [remote states](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/server-e2e.ts#L128). In order to properly ensure they are reinitialized appropriately, we will need to restart the server. Thus, we change `baseUrl`'s `requireRestartOnChange` property to `server`.

### Steps to test


1. `cypress open` a project with a `baseUrl`
2. Execute a test within the project
3. Update the `baseUrl` in cypress.config.js
4. Ensure that the server/browser relaunches and that the url in the browser references the updated `baseUrl`

### How has the user experience changed?


Before

https://user-images.githubusercontent.com/4873279/172271308-5561a2e3-d6a1-41af-8e24-721dc6e70c01.mov

After

https://user-images.githubusercontent.com/4873279/172271323-bcdf97b9-9ed0-4cb6-b749-f030306dc5f1.mov

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
